src/Hint/GHC.hs

```diff
@@ -25,6 +25,10 @@ import DynFlags as X (xFlags, xopt, LogAction, FlagSpec(..))
 import DynFlags as X (xFlags, xopt, LogAction)
 #endif
 
+#if __GLASGOW_HASKELL__ >= 800
+import DynFlags as X ( WarnReason(NoReason) )
```

Comment:
It's more like it was wildly inconsistent before, and I'm trying to make it less so :)